met-web/src/components/engagement/view/EngagementView.tsx

```diff
@@ -27,9 +33,19 @@ export const EngagementView = () => {
         navigate(`/survey/submit/${surveyId}`);
     };
 
+    useEffect(() => {
```

Comment:
seems avoidable.

We can move the modal states (isEmailModalOpen and defaultPanel) to the actionContext. We can pass the setEmailModalOpen and the setDefaultPanel or handle methods that can do that using useContext() to the thankyou panel and we can on close in the success panel set the default panel as we wish

Reply:
You could consider a similar pattern to how the modal was handled here https://github.com/bcgov/met-public/pull/456

Reply:
You can also merge it, and I'll replicate that pattern